camel/memories/context_creators/score_based.py

```diff
@@ -79,14 +78,20 @@ def create_context(
             RuntimeError: If it's impossible to create a valid context without
                 exceeding the token limit.
         """
-        context_units = [
-            _ContextUnit(
-                idx,
-                record,
-                self.token_counter.count_tokens_from_messages(
-                    [record.memory_record.to_openai_message()]),
-            ) for idx, record in enumerate(records)
-        ]
+        # Create unique context units list
```

Comment:
Just curious in which scenario the context can be not unique?

Reply:
Sometimes we will save a message into two different memory blocks. If they are both retrieved (high possibility if you use both vectordb memory and chat history memory) and added to the context, the context length will be wasted.

Reply:
I see, thanks.